### DISCUSSION TITLE
How is this used?

### DISCUSSION THREAD

**Discussion by jonnyijapan** [nishanths/exhaustive#45] [Q&A]
How is this used?
Can it be used such that compilation will fail until all switches are exhaustive?
Is there any IDE integration?

**Comment by nishanths** [comment]
> How is this used?

Usually `exhaustive` is run during development and in CI. During development, it is usually run from the command line. In CI, it is included as part of the source code vet/lint step; if the source code has vetting errors, the build should be failed. 

> Can it be used such that compilation will fail until all switches are exhaustive?

`exhaustive` can't be included as part of compilation, i.e. as part of `go build`. But you can run it separately after your compilation command, like:

```
% go build
% exhaustive
%
```

For convenience, if you use Make, you could have a target `all` that combines these commands:

```Makefile
.PHONY: all
all:
	go build
	go vet
	exhaustive
	go test
```

Then one can just run  `make all` from the command line:

```
% make all
%
```

> Is there any IDE integration?

There isn't a direct IDE integration provided in this repository. But the Go project or other open source projects may provide an IDE integration that works with any [`analysis.Analyzer`](https://pkg.go.dev/golang.org/x/tools/go/analysis), of which `exhaustive` is one. Another option might be the `golangci-lint` project, which may have an IDE integration---`exhaustive` is one of the available linters there.
